### ISSUE TITLE
Odświeżanie tokena autoryzacyjnego

### ISSUE THREAD

**Issue by kdabek** [allegro/allegro-api#9116]
Występuje problem: 
 - [x] z zasobem REST API / with a REST API resource
 - [x] ze środowiskiem produkcyjnym / with the production environment
 - [ ] ze środowiskiem testowym / with the test environment
 - [x] z logowaniem lub autoryzacją / with login or authorization
 - [ ] na portalu https://developer.allegro.pl/ / at https://developer.allegro.pl/en/
 - [ ] inny / other

Witam, mam problem z odświeżaniem tokena autoryzacyjnego dla naszego klienta.
Klient posiada kilka konfiguracji i losowo występuje problem z odświeżaniem tegóż tokena.

Sytuacja z wczoraj (28.05.2024) wygląda tak:
1. Odświeżamy tokeny dla wszystkich konfiguracji po kolei o godzinie [2024-05-28 09:47:06]
2. Otrzymujemy poprawnie odpowiedź i dostajemy nową parę `access_token` + `refresh_token` i je zapisujemy
3. Leci request `POST https://api.allegro.pl/shipment-management/shipments/create-commands` z tym access tokenem powyżej o godzinie [2024-05-28 10:15:07]
4. Dostajemy odpowiedź 401 `Access token expired` [trace-id: d8f24896a3acecf7]

Więc opcje są tutaj dwie: albo w międzyczasie wyszedł od nas ponownie request odświeżania tokenu, chociaż nie mamy tego zalogowanego i odświeżanie następuje co X godzin, no albo z tym tokenem powyżej coś jest jednak nie tak.

**Comment by PrzemyslawLukanowski** [comment]
Witam, powodem wygaśnięcia tokenów byłą operacja z 28.05.24, 10:06 - użytkownik, do którego należał token, zmienił w tym momencie hasło na koncie.

**Comment by Toumash** [comment]
> Witam, powodem wygaśnięcia tokenów byłą operacja z 28.05.24, 10:06 - użytkownik, do którego należał token, zmienił w tym momencie hasło na koncie.

To może wyjasniać dlaczego powoli tracę ważne tokeny uzytkowników : D

Czy są jeszcze jakieś zdarzenia w których tokeny tracą ważność poza poniższymi?
1. Odpięcie aplikacji z konta (usunięcie powiązania)
2. Reset hasła
3. Wygaśnięcie samego tokena (upływ czasu)
4. Użycie refresh_tokena (refresh)

**Comment by PrzemyslawLukanowski** [comment]
@Toumash Jest jeszcze kilka przypadków :) Token również wygaśnie, gdy użytkownik:

- wyloguje się ze wszystkich urządzeń, np. poprzez zakładkę [Logowanie i hasło](https://allegro.pl/moje-allegro/moje-konto/logowanie-i-haslo),
- zmian adres e-mail,
- otrzyma blokadę sprzedaży,
- przekroczy liczbę aktywnych sesji (max. 20 otwartych sesji dla jednego użytkownika).

**Comment by kdabek** [comment]
Aha 😄  więc nie mam czego szukać ;) dziękuję za informację i myślę że warto byłoby umieścić gdzieś w dokumentacji tę informację bo zaoszczędzi to czasu obu stronom

**Comment by PrzemyslawLukanowski** [comment]
Informacja znajduje się w naszym [FAQ](https://developer.allegro.pl/faq#w-jakich-sytuacjach-twoj-token-moze-utracic-waznosc-), dodamy ją również na poziomie [poradnika](https://developer.allegro.pl/tutorials/uwierzytelnianie-i-autoryzacja-zlq9e75GdIR) :)

**Comment by kdabek** [comment]
Tak myślę że poradnik jest bardziej odpowiednim miejscem dla developerów ;)

**Comment by VigorousGreenPanaceum** [comment]
@PrzemyslawLukanowski

Chciałbym jeszcze dopytać o jedną rzecz. Jak najbardziej wyżej w wątku (jak i w poradniku) podane zostały wszystkie możliwe przyczyny wygaśnięcia access tokenu. Brakuje mi natomiast informacji: w których z wymienionych sytuacji wygasa również i refresh token, przez co wymagane będzie również ponowne ręczne uwierzytelnienie użytkownika?

Przykładowo, czy przekroczenie liczby 20 aktywnych sesji powoduje unieważnienie refresh tokenu?

Pozdrawiam

**Comment by Lukasz-Zurek** [comment]
@VigorousGreenPanaceum Tak, unieważniony zostanie zarówno access_token, jak i refresh_token.

**Comment by Lukasz-Zurek** [comment]
@VigorousGreenPanaceum A ja dziękuję Tobie za poruszenie tej kwestii, ponieważ to nie jest takie oczywiste. Wkrótce to zaktualizujemy :)
